azure/regional_baseuri.go

```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/pkg/errors"
 )
 
-type aliasAuth = Authorizer
+type aliasAuth = Authorizer //nolint:unused // This is a false positive since aliasAuth is used right below.
```

Comment:
I am a little confused here, Why can't we use the below here?
```suggestion
type aliasAuth Authorizer
```

Reply:
The relevant check reports this as unused even though it’s used (false positive). I briefly searched for issues related to it in the staticcheck repo and found a few with the mention that they‘ll work on it.

Reply:
Sorry, I did not frame the question right. I was wondering why is `type aliasAuth = Authorizer` and not `type aliasAuth Authorizer`. Looks like it doesn't really matter since `aliasAuth` also turns out to be an _interface_ type in both the scenarios.

Reply:
that's a good question. I tried adjusting the code according to your suggestion and lint doesn't complain anymore (of course also removed the nolint comment) and tests seem to pass too. 

I wonder what the original intention was of adding that type alias though? The [difference is explained quite well on this page](https://yourbasic.org/golang/type-alias/) but I wonder why it's needed at anyway? Why not just work with `Authorizer` itself?

Reply:
should we try just removing it then? possibly it was needed at some point when it was added and that reason no longer exists.

Reply:
I tried to remove it and that led me to a way too long investigation why it doesn't work. 
Summary: the type alias is needed because type `Authorizer` also has a method named `Authorizer()` (same name) and the compiler & linter gets confused when type `baseURIAdapter` wants to embed the interface `Authorizer`

more or less minimum reproducing code: https://go.dev/play/p/nwjcUf9x4DI
how it would work without type alias (rename Authorizer method to Authorizor e.g.): https://go.dev/play/p/A5kCr0FrBPE

I think we can just settle to keep a type alias though. I'll use the other form @nawazkh proposed even though it has a slightly different semantic meaning.

Reply:
code is updated & commented. PTAL @CecileRobertMichon @nawazkh